include/LeMonADE/updater/UpdaterAbstractCreate.h

```diff
@@ -0,0 +1,301 @@
+/*--------------------------------------------------------------------------------
+    ooo      L   attice-based  |
+  o\.|./o    e   xtensible     | LeMonADE: An Open Source Implementation of the
+ o\.\|/./o   Mon te-Carlo      |           Bond-Fluctuation-Model for Polymers
+oo---0---oo  A   lgorithm and  |
+ o/./|\.\o   D   evelopment    | Copyright (C) 2013-2015 by 
+  o/.|.\o    E   nvironment    | LeMonADE Principal Developers
+    ooo                        | 
+----------------------------------------------------------------------------------
+
+This file is part of LeMonADE.
+
+LeMonADE is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+LeMonADE is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with LeMonADE.  If not, see <http://www.gnu.org/licenses/>.
+
+--------------------------------------------------------------------------------*/
+#ifndef LEMONADE_UPDATER_ABSTRACT_CREATE_H
+#define LEMONADE_UPDATER_ABSTRACT_CREATE_H
+
+
+/**
+ * @file
+ *
+ * @class UpdaterAbstractCreate
+ *
+ * @brief abstract updater to create systems
+ * 
+ * @details This abstract class provides the three basic functions to create systems: add a single monomer, add a connected monomer and move the system to find some free space
+ *
+ * @tparam IngredientsType
+ *
+ **/
+
+#include <LeMonADE/updater/AbstractUpdater.h>
+#include <LeMonADE/utility/MonomerGroup.h>
+#include <LeMonADE/utility/DepthIterator.h>
+#include <LeMonADE/utility/Vector3D.h>
+
+template<class IngredientsType>
+class UpdaterAbstractCreate:public AbstractUpdater
+{
+public:
+  UpdaterAbstractCreate(IngredientsType& ingredients_):ingredients(ingredients_){}
+  
+  virtual void initialize();
+  virtual bool execute();
+  virtual void cleanup();
+  
+protected:
+  IngredientsType& ingredients;
+  RandomNumberGenerators rng;
```

Comment:
I would suggest to make this private. Making a new instance of this in a derived class would not make any big difference, and the instance here would anyways be inconvenient to use, because it has to be addressed with the somewhat lengthy syntax
UpdaterAbstractCreate<IngredientsType>::rng